### ISSUE TITLE
Infinite refresh loop on 1.8.2-rc.3

### ISSUE THREAD

**Issue by arggh** [meteor/meteor#10736]
We just updated to `1.8.2-rc.3` and noticed an infinite refresh loop occurring randomly **and often**.

This makes it nearly impossible to actually use this release. After a code change, Meteor just starts rebuilding bundles over and over again, with a 503 in console on page load:

<details>
<summary>stack trace</summary>

```
dynamic-import.js:1 Failed to load resource: the server responded with a status of 503 (Service Unavailable)
guests.js:12 Uncaught TypeError: module.dynamicImport is not a function
    at Route.waitOn (guests.js:12)
    at route.js:263
    at Array.forEach (<anonymous>)
    at Route.waitOn (route.js:262)
    at route._actionHandle (router.js:165)
    at ostrio_flow-router-extra.js?hash=947a4c61a07fb5ea604f2c97cf40c494f6b1b210:3528
    at nextEnter (ostrio_flow-router-extra.js?hash=947a4c61a07fb5ea604f2c97cf40c494f6b1b210:3352)
    at ostrio_flow-router-extra.js?hash=947a4c61a07fb5ea604f2c97cf40c494f6b1b210:3529
    at nextEnter (ostrio_flow-router-extra.js?hash=947a4c61a07fb5ea604f2c97cf40c494f6b1b210:3352)
    at ostrio_flow-router-extra.js?hash=947a4c61a07fb5ea604f2c97cf40c494f6b1b210:3529
```

</details>

A restart seems to fix the issue for a while.

This has never happened to us before, and we've been using both `1.8.1` and `1.8.2-beta.x`. We do have some code in `/client/node_modules/*`, changes to which previously weren't triggering any rebuilds, so that would be my first guess where to look into.

_Might be related:_ https://github.com/meteor/meteor/issues/10257

**Comment by arggh** [comment]
I did a quick test, which indicates that only the code changes in non-root `node_modules` folders (eg. `/client/node_modules/*`) trigger the infinite rebuilds.

**Comment by arggh** [comment]
Reproduction here: https://github.com/arggh/meteor-infinite-rebuild-issue

**Comment by rj-david** [comment]
I normally encounter this when doing `meteor npm install`

It goes away after I restart the build so never dug deeper.

**Comment by arggh** [comment]
**Update:** I just managed to trigger the infinite refresh loop by making a change to a local npm package (declared in `package.json` with `file:../packages/arggh-ui`)

**Comment by benjamn** [comment]
Interestingly, while I can reproduce this with the published `1.8.2-rc.6` release, I can't reproduce it when running Meteor from a checkout, which suggests it's something racy/timing-sensitive.

**Comment by benjamn** [comment]
I believe this will be fixed if you run `meteor update --release 1.8.2-rc.7`, which includes PR #10763. Let us know if you continue to encounter the same problem!

**Comment by arggh** [comment]
Indeed it appears to have been fixed 🎉

This is very much appreciated!
